### PR TITLE
Add action to publish operator images

### PR DESCRIPTION
**Pull request by ShyamsundarR**
Signed-off-by: Shyamsundar Ranganathan <srangana@redhat.com>

### CODE REVIEW

**Comment by ShyamsundarR** [comment]
Note: Only when PRs are pushed (merged) to main branch will images be published.

@Madhu-1 could you take a first look?
- Intention is not to handle branches yet, just main branch pushes

Naming scheme would be (eventually):

- main: (any push to main branch)
    - Image updated: main-canary (only this is supported with this PR)
- vX.Y.Z: (tag created on branch vX.Y)
    - Image updated: vX.Y.Z
- vX.Y: (any push to branch (non-main))
    - Image updated: vX.Y-canary

File: .github/workflows/sanity.yml [hunk 1]
```diff
@@ -127,3 +128,33 @@ jobs:
           make deploy
           kubectl get deployment -n ramen-system
           kubectl -n ramen-system wait --for=condition=Available --timeout=60s deploy/ramen-controller-manager
+
+  publish-image:
+    name: Publish built image
+    needs: [deploy-check, lint, golangci]
```
Comment (Madhu-1) [inline .github/workflows/sanity.yml:134]: Few items in this are not required? i assume PR will get merge only when these tests are passed.

File: .github/workflows/sanity.yml [hunk 2]
```diff
@@ -127,3 +128,33 @@ jobs:
...
+    if: >
```
Comment (Madhu-1) [inline .github/workflows/sanity.yml:135]: this `if` check will get increased if we want to support more branches and tags. I think it's better to move this to a new file.

**Review by Madhu-1 — COMMENTED** [review]
left question/suggestion. else LGTM

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (ShyamsundarR) [reply .github/workflows/sanity.yml:134]: True, but job runs post a push as well, so say there was any auto merge etc. then re-running the jobs and then on success pushing it is safer.

File: .github/workflows/sanity.yml (hunk 2 above)
Comment (ShyamsundarR) [reply .github/workflows/sanity.yml:135]: There is a bit of [scripting tricks](https://github.com/backube/scribe/blob/2dc593bcd4c77e91c8195e9e6820487e53dfc3e5/.github/workflows/operator.yml#L277-L318) that can help, hoping that would ease the conditions in the future. If not we will move this to a separate file.

**Review by Madhu-1 — APPROVED** [review]

**Review by tjanssen3 — APPROVED** [review]
Looks good to me.

**Pull request merged**